lib/Subscribers/SubscriberSaveController.php

```diff
@@ -0,0 +1,218 @@
+<?php
+
+namespace MailPoet\Subscribers;
+
+use MailPoet\CustomFields\CustomFieldsRepository;
+use MailPoet\Entities\StatisticsUnsubscribeEntity;
+use MailPoet\Entities\SubscriberEntity;
+use MailPoet\Entities\SubscriberSegmentEntity;
+use MailPoet\Newsletter\Scheduler\WelcomeScheduler;
+use MailPoet\Segments\SegmentsRepository;
+use MailPoet\Settings\SettingsController;
+use MailPoet\Statistics\Track\Unsubscribes;
+use MailPoet\Util\Security;
+use MailPoet\WP\Functions as WPFunctions;
+
+class SubscriberSaveController {
+  /** @var CustomFieldsRepository */
+  private $customFieldsRepository;
+
+  /** @var Security */
+  private $security;
+
+  /** @var SettingsController */
+  private $settings;
+
+  /** @var SegmentsRepository */
+  private $segmentsRepository;
+
+  /** @var SubscriberCustomFieldRepository */
+  private $subscriberCustomFieldRepository;
+
+  /** @var SubscribersRepository */
+  private $subscribersRepository;
+
+  /** @var SubscriberSegmentRepository */
+  private $subscriberSegmentRepository;
+
+  /** @var Unsubscribes */
+  private $unsubscribesTracker;
+
+  /** @var WelcomeScheduler */
+  private $welcomeScheduler;
+
+  /** @var WPFunctions */
+  private $wp;
+
+  public function __construct(
+    CustomFieldsRepository $customFieldsRepository,
+    Security $security,
+    SettingsController $settings,
+    SegmentsRepository $segmentsRepository,
+    SubscriberCustomFieldRepository $subscriberCustomFieldRepository,
+    SubscribersRepository $subscribersRepository,
+    SubscriberSegmentRepository $subscriberSegmentRepository,
+    Unsubscribes $unsubscribesTracker,
+    WelcomeScheduler $welcomeScheduler,
+    WPFunctions $wp
+  ) {
+    $this->customFieldsRepository = $customFieldsRepository;
+    $this->security = $security;
+    $this->settings = $settings;
+    $this->segmentsRepository = $segmentsRepository;
+    $this->subscriberSegmentRepository = $subscriberSegmentRepository;
+    $this->subscribersRepository = $subscribersRepository;
+    $this->subscriberCustomFieldRepository = $subscriberCustomFieldRepository;
+    $this->unsubscribesTracker = $unsubscribesTracker;
+    $this->welcomeScheduler = $welcomeScheduler;
+    $this->wp = $wp;
+  }
+
+  public function save(array $data): SubscriberEntity {
+    if (is_array($data) && !empty($data)) {
+      $data = WPFunctions::get()->stripslashesDeep($data);
```

Comment:
I wonder why are you using the static call here when WPFunctions are injected into this class.

Reply:
@pavel-mailpoet This method is used in [the old model](https://github.com/mailpoet/mailpoet/blob/master/lib/Models/Subscriber.php#L343-L347)
I expected that there is a good reason. If you think that this shouldn't be here, I can remove it.